### PR TITLE
Fix for 20406. Do not output unnecessary decimal warning

### LINKED ISSUE #20406: false DecimalTypeDefaultWarning[30000] because converter is ignored
I'm using a **converter** on a `decimal` property which converts the value to `Int64` (BigInt on SqlServer) but I still get a warning logged (to Seq) on each app start (first context creation) for every single decimal property being part of the model (all have the converter assigned):

> No type was specified for the decimal column 'Balance' on entity type 'Customer'. This will cause values to be silently truncated if they do not fit in the default precision and scale. Explicitly specify the SQL server column type that can accommodate all the values using 'HasColumnType()'.

(does not happen when using SQLite provider)

### Steps to reproduce

A property somewhere in the model:
``` C#
public decimal Balance { get; set; }
```
The converter applied to `ModelBuilder`
``` C#
modelBuilder.Entity<Customer>()
    .Property(c => c.Balance).ConvertToIntegerMoneyCents();
```
The converter itself:
``` C#
private static readonly ValueConverter decimalMoneyToLongCentsConverter = new DecimalMoneyToLongCentsConverter();

public static PropertyBuilder<decimal> ConvertToIntegerMoneyCents(this PropertyBuilder<decimal> propertyBuilder, string suffix = "Cents")
=> propertyBuilder
	.HasConversion(decimalMoneyToLongCentsConverter)
	.HasColumnName(propertyBuilder.Metadata.Name + suffix);

private sealed class DecimalMoneyToLongCentsConverter : ValueConverter<decimal, long>
{
	private static readonly Expression<Func<decimal, long>> convertToProviderExpression = d => TryConvertToProvider(d);
	private static readonly Expression<Func<long, decimal>> convertFromProviderExpression = l => l / 100m;

	internal DecimalMoneyToLongCentsConverter()
		: base(convertToProviderExpression, convertFromProviderExpression)
	{
	}

	private static long TryConvertToProvider(decimal value)
	{
		var cents = value * 100;
		if ( decimal.Truncate(cents) != cents )
			throw new ArgumentException("Using money with more than 2 digits after comma would loose information!", nameof(value));
		return (long)cents;
	}
}
```

I could add `.HasColumnType("BigInt")` but that would break the currently provider-agnostic definition of the model. (While this might work for SQLite, for other providers this will throw an error.)

### Further technical details

EF Core version: 3.1.1
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1.2
Operating system: W10
IDE: VS 16.5.1

### PR DESCRIPTION
**Pull request by lajones**
Do not output the SQL Server decimal warning if the user has specified a value converter.

Fixes #20406

### CODE REVIEW

File: src/EFCore.SqlServer/Internal/SqlServerModelValidator.cs [hunk 1]
```diff
@@ -74,12 +74,15 @@ public override void Validate(IModel model, IDiagnosticsLogger<DbLoggerCategory.
...
-                        && ConfigurationSource.Convention.Overrides(typeMappingConfigurationSource))
-                    || (typeConfigurationSource != null
-                        && ConfigurationSource.Convention.Overrides(typeConfigurationSource)))
+                var valueConverterConfigurationSource = (property as IConventionProperty)?.GetValueConverterConfigurationSource();
+                if ((columnTypeConfigurationSource == null
+                        && ConfigurationSource.Convention.Overrides(typeMappingConfigurationSource)
+                        && ConfigurationSource.Convention.Overrides(valueConverterConfigurationSource))
+                    || (columnTypeConfigurationSource != null
+                        && ConfigurationSource.Convention.Overrides(columnTypeConfigurationSource)
+                        && ConfigurationSource.Convention.Overrides(valueConverterConfigurationSource)))
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Internal/SqlServerModelValidator.cs:85]: This is not enough, we should also check whether the value converter has a provider type different from `decimal`

Comment (lajones) [reply src/EFCore.SqlServer/Internal/SqlServerModelValidator.cs:85]: @AndriySvyryd You mean a decimal to decimal converter?

Comment (AndriySvyryd) [reply src/EFCore.SqlServer/Internal/SqlServerModelValidator.cs:85]: Yes

Comment (lajones) [reply src/EFCore.SqlServer/Internal/SqlServerModelValidator.cs:85]: OK. I guess it's possible. I'll update.

Comment (lajones) [reply src/EFCore.SqlServer/Internal/SqlServerModelValidator.cs]: Done

**Pull request merged**